### PR TITLE
Add multiple interfaces support

### PR DESCRIPTION
**Pull request by morpheu**
Add multiple network ids or networks comma separated values including option to specify private and public interface index.

### CODE REVIEW

File: cloudstack.go [hunk 1]
```diff
@@ -696,30 +711,46 @@ func (d *Driver) setDiskOffering(diskOffering string, diskOfferingID string) err
...
 
 	cs := d.getClient()
 	var network *cloudstack.Network
 	var err error
-	if d.NetworkID != "" {
-		network, _, err = cs.Network.GetNetworkByID(d.NetworkID, d.setParams)
+	var networkIDs, networkNames []string
+
+	if networkID != "" {
+		networkIDs = make([]string, len(strings.Split(networkID, ",")))
```
Comment (cezarsa) [inline cloudstack.go]: You should call Split only once and reuse the result here and in the range statement.

File: cloudstack.go [hunk 2]
```diff
@@ -696,30 +711,46 @@ func (d *Driver) setDiskOffering(diskOffering string, diskOfferingID string) err
...
 	}
 
-	d.NetworkID = network.Id
-	d.Network = network.Name
+	d.NetworkID = networkIDs
+	d.Network = networkNames
 
-	log.Debugf("network id: %q", d.NetworkID)
-	log.Debugf("network name: %q", d.Network)
+	log.Debugf("network ids: %q", d.NetworkID)
```
Comment (cezarsa) [inline cloudstack.go]: Use `%v`, `%q` is only for strings.

File: cloudstack.go [hunk 3]
```diff
@@ -944,8 +975,12 @@ func (d *Driver) associatePublicIP() error {
 	log.Infof("Associating public ip address...")
 	p := cs.Address.NewAssociateIpAddressParams()
 	p.SetZoneid(d.ZoneID)
-	if d.NetworkID != "" {
-		p.SetNetworkid(d.NetworkID)
+	if len(d.NetworkID) > 0 {
+		if d.PublicInterfaceIndex < len(d.NetworkID) {
```
Comment (cezarsa) [inline cloudstack.go]: You could invert the condition and remove the `else`.

File: cloudstack.go (hunk 2 above)
Comment (morpheu) [reply cloudstack.go]: Done!

File: cloudstack.go (hunk 3 above)
Comment (morpheu) [reply cloudstack.go]: Done!

**Pull request merged**